### PR TITLE
feature: generic smtp provider for email notifications

### PR DESCRIPTION
**Pull request by lucasmenendez**
This PR replaces the current email notification service from the Sendgrid API with a generic SMTP implementation, allowing the email provider to be changed by simply updating envvars.

Some new code has been commented to be uncommented when #14 is merged.

### CODE REVIEW

File: notifications/smtp/smtp.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	SMTPUsername string
+	SMTPPassword string
+}
+
+type SMTPEmail struct {
+	config *SMTPConfig
+	auth   smtp.Auth
+}
+
+func (se *SMTPEmail) Init(rawConfig any) error {
```
Comment (p4u) [inline notifications/smtp/smtp.go:29]: Comments please, at least on all exported methods and types.

Comment (p4u) [inline notifications/smtp/smtp.go:29]: Nit: Use a constructor function instead of a Init() metod.

File: notifications/smtp/smtp.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		return fmt.Errorf("invalid SMTP configuration")
+	}
+	// parse from email
+	if _, err := mail.ParseAddress(config.FromAddress); err != nil {
+		return fmt.Errorf("could not parse from email: %v", err)
+	}
+	// set configuration in struct
+	se.config = config
+	// init SMTP auth
+	se.auth = smtp.PlainAuth("", se.config.SMTPUsername, se.config.SMTPPassword, se.config.SMTPServer)
```
Comment (p4u) [inline notifications/smtp/smtp.go:42]: The SMTP server could allow no authentication (i.e for local testing using local sendmail). So auth should be only configured if user/password are defined, and they should not be enforced to exist.

File: notifications/smtp/smtp.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+type SMTPEmail struct {
+	config *SMTPConfig
+	auth   smtp.Auth
+}
+
+func (se *SMTPEmail) Init(rawConfig any) error {
+	// parse configuration
+	config, ok := rawConfig.(*SMTPConfig)
```
Comment (p4u) [inline notifications/smtp/smtp.go:31]: Umm, why rawConfig is any? Is this implementing an interface?

**Review by p4u — CHANGES_REQUESTED** [review]

File: notifications/smtp/smtp.go (hunk 3 above)
Comment (lucasmenendez) [reply notifications/smtp/smtp.go:31]: yes it is an interface

File: notifications/smtp/smtp.go (hunk 2 above)
Comment (lucasmenendez) [reply notifications/smtp/smtp.go:42]: Updated with new methods that are also required for testing.

**Review by p4u — APPROVED** [review]

**Pull request merged**